### PR TITLE
Attempt to fix console. (#4294)

### LINKED ISSUE #4290: Web3/Parity Console does not work at all
screenshot from 2017-01-24 16 14 49

`s/window.parity.api/window.api/g` should fix it.

### PR DESCRIPTION
**Pull request by tomusdrw**
* Attempt to fix console.
* Fixing couple of possible exceptions

Closes #4290

### CODE REVIEW

File: js/src/dapps/static/console.js [hunk 1]
```diff
@@ -17,15 +17,19 @@
...
+  window.api = window.parity.api;
+
+if (typeof(window.api) === 'object')
+	window.api.subscribe('eth_blockNumber', function (error, blockNumber) {
+	    if (error) {
+	    	console.log('error', error);
+	    	return;
+	    }
+	    refreshWatches();
+	});
```
Comment (jacogr) [inline js/src/dapps/static/console.js]: Indentation of this block looks very random, like mixing on tabs/spaces. (Pre-linting, visual check)

File: js/src/dapps/static/console.js [hunk 2]
```diff
@@ -17,15 +17,19 @@
...
-if (typeof(window.parity) == 'object')
-  window.api = window.parent.secureApi;
-  window.parity.api.subscribe('eth_blockNumber', function (error, blockNumber) {
-    if (error) {
-      console.log('error', error);
-      return;
-    }
-    refreshWatches();
-  });
+if (typeof(window.parent.secureApi) == 'object')
```
Comment (jacogr) [inline js/src/dapps/static/console.js]: Should probably be `===` (along with the one below)

File: js/src/dapps/static/console.js [hunk 3]
```diff
@@ -56,21 +60,42 @@ function evaluate(x) {
...
+  var toString = Object.prototype.toString;
   if (x === undefined)
     return '<span class="undefinedType">undefined</span>';
   if (x === null)
     return '<span class="undefinedType">null</span>';
   if (typeof(x) == "string")
     return `"<span class="${typeof(x)}Type">${escapeHtml(x)}</span>"`;
-  if (Object.prototype.toString.call(x) === '[object Array]')
-    return `[${x.map(displayReady).join(', ')}]`;
+  if (toString.call(x) === '[object Array]')
```
Comment (jacogr) [inline js/src/dapps/static/console.js]: `Array.isArray` ?

File: js/src/dapps/static/console.js [hunk 4]
```diff
@@ -159,7 +184,7 @@ function exec() {
       let res;
       try {
         res = evaluate(c);
-        if (typeof(res) == 'object' && res !== null && res.constructor.name == "Promise") {
+        if (typeof(res) == 'object' && res !== null && res.constructor && res.constructor.name == "Promise") {
```
Comment (jacogr) [inline js/src/dapps/static/console.js]: `===`

Be aware that `res.constructor.name` may not work in FF with the es6-promise polyfills. (Have not checked myself with this branch, but Promise checking this way is always a bit of an issue)

**Comment by jacogr** [comment]
Some minor grumbles.

**Comment by tomusdrw** [comment]
I've noticed some other issues with console app (it works in development mode, but fails in release). Will prepare a fix for master first (addressing those issues) and then re-open this.

**Pull request closed without merging**